### PR TITLE
refactor: be optimistic to empty reverse result

### PR DESCRIPTION
**Pull request by UncleBill**
silence react-query's complaining about undefined result 

MF-0000

### CODE REVIEW

File: packages/web3-hooks/base/src/useReverseAddress.ts [hunk 1]
```diff
@@ -5,13 +5,12 @@ import { useWeb3State } from './useWeb3State.js'
...
-    return useQuery<string | undefined>({
+    return useQuery({
         queryKey: ['reverse', address],
-        enabled: !isBad,
+        enabled: !!NameService?.reverse,
         queryFn: async () => {
-            if (isBad) return
-            return NameService.reverse?.(address)
+            if (!address) return null
+            return NameService!.reverse!(address) || null
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useReverseAddress.ts:13]:
```suggestion
            return NameService?.reverse?.(address) || null
```

File: packages/web3-hooks/base/src/useReverseAddress.ts [hunk 2]
```diff
@@ -5,13 +5,12 @@ import { useWeb3State } from './useWeb3State.js'
...
-    const isBad = !address || !NameService
-    return useQuery<string | undefined>({
+    return useQuery({
         queryKey: ['reverse', address],
-        enabled: !isBad,
+        enabled: !!NameService?.reverse,
         queryFn: async () => {
-            if (isBad) return
-            return NameService.reverse?.(address)
+            if (!address) return null
```
Comment (guanbinrui) [inline packages/web3-hooks/base/src/useReverseAddress.ts:12]: Why not just return `undefined`? It seems to cause unnecessary changes.

Comment (UncleBill) [reply packages/web3-hooks/base/src/useReverseAddress.ts:12]: react-query won't cache `undefined` result, it will try fetching again until non `undefined` result.

> It seems to cause unnecessary changes.

Those changes are typing only, no runtime change.

File: packages/web3-hooks/base/src/useReverseAddress.ts (hunk 1 above)
Comment (UncleBill) [reply packages/web3-hooks/base/src/useReverseAddress.ts:13]: `enabled: !!NameService?.reverse,` makes sure it's not undefined.

**Review by guanbinrui — APPROVED** [review]

**Comment by UncleBill** [comment]
It has reduced a lot of requests.
